src/recovery.c

```diff
@@ -684,3 +687,63 @@ void rec_restore_mode(int (*restore_mode)(FILE *file))
 
 	rec_restoring_now = 0;
 }
+
+uint64_t rec_read_cands(char *session)
+{
+	char line[LINE_BUFFER_SIZE];
+	char *other_name;
+	FILE *other_file = NULL;
+	int index, argc;
+
+	if (!john_main_process && options.node_min) {
+		char suffix[1 + 20 + sizeof(RECOVERY_SUFFIX)];
+
+		sprintf(suffix, ".%u%s", options.node_min, RECOVERY_SUFFIX);
+		other_name = path_session(session, suffix);
+	} else {
+		other_name = path_session(session, RECOVERY_SUFFIX);
+	}
+
+	if (!strcmp(other_name, rec_name))
+		error_msg("New session name can't be same as catchup session\n");
+
+	if (!(other_file = fopen(other_name, "r+")))
```

Comment:
Maybe this should be `r` and not `r+`? I wasn't quite aware of the meaning of `+` when I copied the code from `rec_restore_args()`. We do lock the file until we exit but we're not supposed to write to it.